pods.go

```diff
@@ -25,25 +25,14 @@ func (c *Client) CreatePod(ctx context.Context, pod *api.Pod) (*api.Pod, error)
 	if err := json.NewEncoder(&podJSON).Encode(pod); err != nil {
 		return nil, fmt.Errorf("failed to encode pod in json: %v", err)
 	}
-	postURL := c.podURL(pod.Namespace)
-	req, err := http.NewRequest("POST", postURL, &podJSON)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create request: POST %q : %v", postURL, err)
-	}
-	res, err := ctxhttp.Do(ctx, c.Client, req)
-	if err != nil {
-		return nil, fmt.Errorf("failed to make request: POST %q: %v", postURL, err)
-	}
-	body, err := ioutil.ReadAll(res.Body)
-	res.Body.Close()
+
+	apiResult, err := CreateKubeResource(ctx, &PodResource{c.Host, pod.Namespace, ""}, podJSON, c.Client)
```

Comment:
I ask because it doesn't seem like it necessarily needs to be. It could also easily be a method on kube client and you wouldn't need to pass in the client each time.